services/repository/src/main/java/org/sagebionetworks/auth/HttpAuthUtil.java

```diff
@@ -136,4 +145,32 @@ public static void reject(HttpServletResponse resp, String reason, HttpStatus st
 		}
 		resp.getWriter().flush();
 	}
+
+	/**
+	 * OAuth errors don't have the same JSON format as our regular error, so this method may be used for
+	 * sending error responses caused by OAuth services
+	 */
+	public static void rejectWithOAuthError(HttpServletResponse resp, OAuthErrorCode errorCode, String description, HttpStatus status) throws IOException {
+		resp.setStatus(status.value());
+
+		// This header is required according to RFC-2612
+		// See: http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.2
+		//      http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.47
+		//      http://www.ietf.org/rfc/rfc2617.txt
+		resp.setContentType("application/json");
+		resp.setHeader("WWW-Authenticate", "\"Digest\" your email");
```

Comment:
check whether WWW-Authenticate is needed.  I think it's specifically for a 401 response